### PR TITLE
[OIS/example] Sharing lock-app cluster service between Linux and OIS platforms

### PR DESCRIPTION
**Pull request by ATmobica**
The OIS implementation of lock-app uses the same lock-app clusters service source code as the Linux platform.
To be always up-to-date with changes it will be worth just sharing these sources.

In the first commit, we create a common lock-app cluster service by moving the lock endpoint, manager and ZCL callbacks from Linux to the lock-common directory. The changes were also applied to Linux lock-app implementation.
The second commit adapts OIS lock-app implementation to common sources usage.

Additionally, we changed "setting door lock state" logs to progress level to be informed of the lock status change directly.
It's useful for integration tests for the OIS platform.

These changes are required for further work on the integration of the certification tests suite for the OIS platform.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]
Note https://github.com/project-chip/connectedhomeip/pull/24532 which will need to be merged with this, in one direction or the other.

**Comment by ATmobica** [comment]
> Note #24532 which will need to be merged with this, in one direction or the other.

Sure, I suppose it will be easier to merge #24532 first. But if it will take more than 2 weeks I'll aks you to merge this PR first. We'll need it for further work on the certification test suite for OIS.

**Comment by bzbarsky-apple** [comment]
@ATmobica I don't know that the order matters much here, for what it's worth.  The rebase is clean in either direction, afaict.

**Comment by ATmobica** [comment]
@andy31415 @bzbarsky-apple PR is up-to-date.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by ATmobica** [comment]
@andy31415 Can we go forward with this PR?

**Comment by ATmobica** [comment]
@andy31415 @bzbarsky-apple we really need this PR to go further, can we go further with it?

**Comment by ATmobica** [comment]
> @ATmobica You need a non-Apple review, right? Have you tried pinging people on the development Slack channel?

@pan- Can you ping someone on the Matter Slack channel?

File: src/test_driver/openiotsdk/integration-tests/lock-app/test_app.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def test_lock_ctrl(device, controller):
                                                                  LOCK_CTRL_TEST_PIN_CODE), requestTimeoutMs=1000)
     assert err == 0
 
-    ret = device.wait_for_output("Setting door lock state to \"Locked\" [endpointId={}]".format(LOCK_CTRL_TEST_ENDPOINT_ID))
+    ret = device.wait_for_output("setting door lock state to \"Locked\"".format(LOCK_CTRL_TEST_ENDPOINT_ID))
```
Comment (pan-) [inline src/test_driver/openiotsdk/integration-tests/lock-app/test_app.py:154]: `.format` can be removed if there is nothing to format.

**Comment by ATmobica** [comment]
@andy31415 Could it be already merged?

**Comment by ATmobica** [comment]
@andy31415 @bzbarsky-apple Can we merge this? This keeps blocking the next changes we would like to deliver.

**Comment by bzbarsky-apple** [comment]
@ATmobica Needs one non-Apple review, right?

**Comment by ATmobica** [comment]
> @ATmobica Needs one non-Apple review, right?

@bzbarsky-apple Probably yes. I need a merge of these changes the most.

**Review by arkq — APPROVED** [review]

**Pull request merged**